great_expectations/expectations/expectation.py

```diff
@@ -1826,7 +1826,7 @@ class ColumnMapExpectation(BatchExpectation, ABC):
     """  # noqa: E501
 
     column: StrictStr = Field(min_length=1, description=COLUMN_DESCRIPTION)
-    mostly: Mostly = 1.0
+    mostly: Mostly = 1.0  # type: ignore[assignment] # FIXME
```

Comment:
I haven't had time to dig into what's going here with `Mostly`
But changing the mypy target to Python 3.9 surfaces this issue where the `Mostly = Annotated[_Mostly, float]` no long sees these `float` types as valid.

My initial assumption is that it's because `Annotated` wasn't being used quite right but I don't know for sure. I tried updating our compatibility layer to first use the `typing_extensions` version before falling back to the stdlib version since `typing_extensions` can include bug fixes or enhancements. But that didn't seem to do anything.